lib/functions.py

```diff
@@ -417,41 +390,31 @@ def filter_pattern(doc_identifier):
             return 'numbers'
     return None
 
-def filter_chapter(doc, language, session):
-    if session['cancellation_requested']:
-        stop_and_detach_tts()
-        raise ValueError('Cancel requested')
-
+def filter_chapter(doc, language):
     soup = BeautifulSoup(doc.get_body_content(), 'html.parser')
     # Remove scripts and styles
     for script in soup(["script", "style"]):
         script.decompose()
-
     # Normalize lines and remove unnecessary spaces
     text = re.sub(r'(\r\n|\r|\n){3,}', '\r\n', soup.get_text().strip())
     text = replace_roman_numbers(text)
     lines = (line.strip() for line in text.splitlines())
     chunks = (phrase.strip() for line in lines for phrase in line.split("  "))
     text = '\n'.join(chunk for chunk in chunks if chunk)
     text = text.replace('»', '"').replace('«', '"')
-
     # Pattern 1: Add a space between UTF-8 characters and numbers
     text = re.sub(r'(?<=[\p{L}])(?=\d)|(?<=\d)(?=[\p{L}])', ' ', text)
-
     # Pattern 2: Split numbers into groups of 4
     text = re.sub(r'(\d{4})(?=\d)', r'\1 ', text)
-
-    chapter_sentences = get_sentences(text, language, session)
+    chapter_sentences = get_sentences(text, language)
     return chapter_sentences
 
-def get_sentences(sentence, language, session, max_pauses=9):
+def get_sentences(sentence, language, max_pauses=9):
     max_length = language_mapping[language]['char_limit']
     punctuation = language_mapping[language]['punctuation']
+    sentence = sentence.replace(".", ";\n")
     parts = []
     while len(sentence) > max_length or sum(sentence.count(p) for p in punctuation) > max_pauses:
-        if session['cancellation_requested']:
-            stop_and_detach_tts()
-            raise ValueError('Cancel requested')
         # Step 1: Look for the last period (.) within max_length
         possible_splits = [i for i, char in enumerate(sentence[:max_length]) if char == '.']    
```

Comment:
...and then look for periods in the text you just removed all periods from.

Reply:
good catch, I'm going to fix it, thanks